src/Simplify.elm

```diff
@@ -3713,6 +3720,35 @@ listIndexedMapChecks { lookupTable, fnRange, firstArg } =
                     []
 
 
+listSumChecks : CheckInfo -> List (Error {})
+listSumChecks checkInfo =
+    case Node.value checkInfo.firstArg of
+        Expression.ListExpr [] ->
+            [ Rule.errorWithFix
+                { message = "Using List.sum on [] will result in []"
+                , details = [ "You can replace this call by []." ]
+                }
+                checkInfo.fnRange
+                [ Fix.replaceRangeBy checkInfo.parentRange "[]" ]
```

Comment:
This should also be 0

Reply:
absolutely right

Reply:
To catch these "dumb mistakes", there could be checks on whether the simplified version doesn't have more narrow types for anything... Which is overkill and needs elm type inference to be complete

Reply:
Yeah. In an ideal world, we would have the Elm compiler compiling the code. Maybe with `Elm-in-Elm` or something? :sweat_smile: